include/bout/petsc_interface.hxx

```diff
@@ -211,6 +343,20 @@ public:
     return *this;
   }
 
+  PetscVector<T>& operator=(const T& f) {
+    BOUT_FOR_SERIAL(i, indexConverter->getRegionAll()) {
+      const PetscInt ind = indexConverter->getGlobal(i);
+      if (ind != -1) {
+        // TODO: consider how VecSetValues() could be used where there
+        // are continuous stretches of field data which should be
+        // copied into the vector.
+        VecSetValue(*vector, ind, f[i], INSERT_VALUES);
```

Comment:
I tried this for the Hypre wrapper and it worked. No idea on performance:

```cpp
    for (const auto& block : region.getBlocks()) {
      std::vector<HYPRE_Int> indices;
      indices.reserve(block.second.ind - block.first.ind);
      for (auto index = block.first; index < block.second; ++index) {
        indices.emplace_back(static_cast<HYPRE_BigInt>(indexConverter->getGlobal(index)));
       }
      HYPRE_IJVectorSetValues(hypre_vector,
                              static_cast<HYPRE_Int>(block.second.ind - block.first.ind),
                              indices.data(), &f[block.first]);
     }
```

Reply:
Yes, I was thinking something along these lines could work. I'd prefer to leave it for later, potentially after some benchmarking however.

Reply:
Yes, that's wise